Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -560,8 +560,13 @@ Color Image::color_at(Gfx::IntPoint const& point) const
 {
     Color color;
     for (auto& layer : m_layers) {
-        if (layer.is_visible() && layer.rect().contains(point))
-            color = color.blend(layer.bitmap().get_pixel(point));
+        if (!layer.is_visible() || !layer.rect().contains(point))
+            continue;
+
+        auto layer_color = layer.bitmap().get_pixel(point);
+        float layer_opacity = layer.opacity_percent() / 100.0f;
+        layer_color.set_alpha((u8)(layer_color.alpha() * layer_opacity));
+        color = color.blend(layer_color);
     }
```

Comment:
Looks good to me and works as expected, but please squash this with the previous commit :)

Reply:
Thanks for the feedback!